packages/p2p/src/components/modal-manager/modal-manager-context-provider.tsx

```diff
@@ -90,24 +135,24 @@ const ModalManagerContextProvider = props => {
 
         if (isDesktop()) {
             if (should_hide_all_modals) {
-                setPreviousModal({});
-                setActiveModal({});
+                setPreviousModal(null);
+                setActiveModal(null);
                 setIsModalOpen(false);
             } else if (previous_modal) {
                 setActiveModal(previous_modal);
-                setPreviousModal({});
+                setPreviousModal(null);
             } else {
-                setActiveModal({});
+                setActiveModal(null);
                 setIsModalOpen(false);
             }
-        } else if (Object.keys(stacked_modal).length !== 0) {
+        } else if (stacked_modal && Object.keys(stacked_modal).length !== 0) {
             if (should_hide_all_modals) {
-                setActiveModal({});
+                setActiveModal(null);
                 setIsModalOpen(false);
             }
-            setStackedModal({});
+            setStackedModal(null);
         } else {
-            setActiveModal({});
+            setActiveModal(null);
             setIsModalOpen(false);
         }
     };
```

Comment:
can you check if this would work? this could reduce the code by a bit

```suggestion
if (isDesktop()) {
  if (should_hide_all_modals || previous_modal) {
    setActiveModal(previous_modal || null);
    setPreviousModal(null);
    setIsModalOpen(!should_hide_all_modals);
  }
} else if (stacked_modal && Object.keys(stacked_modal).length !== 0) {
  if (should_hide_all_modals) {
    setActiveModal(null);
    setIsModalOpen(false);
  }
  setStackedModal(null);
} else {
  setActiveModal(null);
  setIsModalOpen(false);
}
```